doc/adr/0002-use-meta-yaml-to-track-feedstock-metadata.md

```diff
@@ -31,13 +31,29 @@ The format and contents of this file are specified as follows.
 ### Top Level Data
 
 ```yaml
-id: noaa-oisst  # top-level ID for the feedstock. must be unique and must match feedstock repo name
+id: noaa-oisst
+version: "1.0" 
 title: "NOAA Optimum Interpolated SST"
 description: "Analysis-ready Zarr datasets derived from NOAA OISST NetCDF"
 pangeo_forge_recipes_version: "0.1"
-pangeo_forge_metadata_spec_version: "1"
+pangeo_forge_metadata_spec_version: "2"
 ```
 
+| name | description | valid characters |
+| -- | -- | -- |
+| id | top-level ID for the feedstock. must be unique and must match feedstock repo name | lower case ascii alpha-numeric, no spaces, only dashes allowed |
+| version | feedstock version | `{MAJOR}.{̨MINOR}` |
+| description | - | Unicode |
+
+#### Version
+
+We use a two-element version of semantic versioning: `MAJOR.MINOR`.
+The recipe version starts at 1.0.
+Updates to the minor version number are considered minor changes;
+when a new minor version is released, bakeries will _overwrite the previous minor version_ of the feedstock's datasets.
+Instead, the bakery should _always retain the latest major version_.
+Updates that involve concatenation of new elements (e.g. extend the temporal range) are considered minor version updates.
+
```

Comment:
I like that an example is given for minor version change: extended temporal range.

What would be an example (or examples) of major version changes? I'm guessing:

- Changing underlying bytes for existing variable arrays (within an existing temporal range)
- Adding variable arrays that didn't exist in prior version
- ... any others?

What about changing metadata such as variable names? That could break user code, but isn't too ... "major" in the colloquial sense?

Reply:
_Adding_ new variables would be minor IMO.

Changing variable names would be a major change, because it would not be backwards compatible.

Changing the processing functions would probably be major.

Updating metadata, e.g. to correct units, descriptions, etc. would be minor.

We should add a table of these to the ADR.

Reply:
```suggestion
Changes are _minor_ if they will be backwards-compatible with user code which relies on the previous version of the dataset. Changes are _major_ if they have the potential to break existing user code. The following table provides examples for each category:

| Change | Minor | Major |
| -------- | ----- | ------ |
| Updating metadata (e.g. to correct units, descriptions, etc.) | :ballot_box_with_check:  |  |
| Adding new variables or timesteps | :ballot_box_with_check: |  |
| Changing existing variable names | | :ballot_box_with_check: |
| Changing preprocessing functions (that alter existing data) | | :ballot_box_with_check: |

```

Preview of the rendered table:

| Change | Minor | Major |
| -------- | ----- | ------ |
| Updating metadata (e.g. to correct units, descriptions, etc.) | :ballot_box_with_check:  |  |
| Adding new variables or timesteps | :ballot_box_with_check: |  |
| Changing existing variable names | | :ballot_box_with_check: |
| Changing preprocessing functions (that alter existing data) | | :ballot_box_with_check: |